pjsip/src/pjsip/sip_transport_tcp.c

```diff
@@ -971,6 +994,14 @@ static pj_status_t lis_create_transport(pjsip_tpfactory *factory,
     }
 
     if (tcp->has_pending_connect) {
+        /* Start the connection timeout timer. */
```

Comment:
Not sure of the locking mechanisms here, but could there be a (very) small window here - i.e.

-  Connection is initiated
-  We come in here because has_pending_connect is set
-  Connection completes before we set the timer
-  Timer eventually expires, bringing the connection down unnecessarily
Would it be technically cleaner to start the timer before starting the connection and then cancelling it afterwards if pending_connect isn't set?
Alternatively, you could put a backstop in tcp_connect_timer (see below)